add prometheus resource utilization

memodi opened the pull request:
adding prometheus resource utilization metics

memodi commented:
/hold

jotak commented on scripts/queries/netobserv_prometheus_queries.yaml:
```diff
@@ -94,6 +101,10 @@
   metricName: ebpfFlowsDroppedTotals
 
+# eBPF flow drops ratio to number of evicted flows in 1 min
+- query: sum(rate(netobserv_agent_dropped_flows_total[1m])) / sum(rate(netobserv_agent_evicted_flows_total)[1m]) > 0.05 or on vector(0)
```
there's a trick here, `netobserv_agent_dropped_flows_total` might sometimes increase for dropped flows *before they are aggregated* (normally, an unaggregated flow is merged with existing flows having the same key) ; which means that this ratio could go up very easily. But I don't have something better to suggest (we might have to create a new metric to "fix" that, in the agent) ... so we can start with that anyway

memodi replied:
@jotak - what unaggregated flows would make `netobserv_agent_dropped_flows_total` counter to increase? because all flows eventually start as unaggregated later to be aggregated? Another metric that's published is `netobserv_agent_evicted_packets_total`  but nothing for packets dropped not sure if that's useful here.

jotak commented on scripts/queries/netobserv_prometheus_queries.yaml:
```diff
@@ -94,6 +101,10 @@
 - query: sum(rate(netobserv_agent_dropped_flows_total[1m]))
   metricName: ebpfFlowsDroppedTotals
 
+# eBPF flow drops ratio to number of evicted flows in 1 min
```
```suggestion
# eBPF flow drops ratio to number of evicted flows in 1 min. Note that netobserv_agent_dropped_flows_total may count unaggregated flows, which might result in a high ratio not representative of the actual (aggregated) flows being dropped.
```

@memodi a comment suggestion , to not forgot about that when we come back later

memodi commented on scripts/queries/netobserv_touchstone_statistics_config.json:
```diff
@@ -116,7 +116,7 @@
                 "aggregations": {
                     "value": [
-                        "max"
+                        "avg"
```
Good point, it's indeed former - taking the "max"/total flows processed value between start and end times. So having "max" here for this one would make sense? unless if we'd want to modify this query to have `rate` function here  for query to be like : `sum(rate(netobserv_ingest_flows_processed[1m])` if we're interested in `avg` flows processed?

jotak replied:
Yes but I think it's not only this one but everywhere where max was used before we should actually keep max, they're all in the same situation afaict, all being counters for which we need to keep only the latest value:
- nWorkloadFlowsProcessedTotals
- lokiRecordsWritten
- lokiRecordsDropped
- ebpfFlowsDroppedTotals
- lokiStorageUsage
- kafkaStorageUsage

memodi replied:
thanks, I reverted to max for those metrics and kept avg for others.

memodi commented:
/verify-owners

memodi commented:
/unhold

memodi commented:
@jotak @Amoghrd - can you take one more pass at this PR and see if it looks good to you?  (@jotak -  particularly if those [prom. queries](https://github.com/openshift-qe/ocp-qe-perfscale-ci/pull/631/files#diff-d4aecea9a2054eeedb7b1d8b5bc450780d846c4757196d6f1bbfa3e5e6cedeaeR551-R569) look good)


One notable change since the last review was: https://github.com/openshift-qe/ocp-qe-perfscale-ci/pull/631/files#diff-e6ffa5dc854b843b3ee3c3c28f8eae2f436c2df2b1ca299cca1fa5982e390cf8R1140-R1166 , where some pipeline refactoring had to be done, more refactoring will follow in https://issues.redhat.com/browse/NETOBSERV-1791

memodi commented:
/verify-owners

memodi commented:
/verify-owners

memodi commented:
/verify-owners

memodi commented:
/verify-owners

memodi commented:
/verify-owners

Amoghrd commented:
/lgtm
